modules/redis/src/main/scala/zio/redis/RedisSubscription.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.redis
+
+import zio._
+
+trait RedisSubscription extends api.Subscription
```

Comment:
Just to clarify, are you using the `RedisSubscription` to ensure clients cannot call other methods while in the subscription regime?

Reply:
That was my intended design. However, the current implementation does not block other commands, but instead requires access through a subscription API to interact with the implementation.